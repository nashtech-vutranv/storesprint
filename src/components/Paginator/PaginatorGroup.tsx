import {useState, FC, KeyboardEvent, ChangeEvent} from 'react'
import {useTranslation} from 'react-i18next'
import {
  Paginator,
  PaginatorProps,
  PaginatorPageState,
  PaginatorCurrentPageReportOptions,
} from 'primereact/paginator'
import {Button} from 'primereact/button'
import TemplatePaginator from './PaginatorTemplate'

interface IPaginator extends PaginatorProps {}

const PaginateGroup: FC<IPaginator> = () => {
  const {t} = useTranslation()

  const [basicFirst, setBasicFirst] = useState(0)
  const [basicRows, setBasicRows] = useState(10)
  const [customFirst1, setCustomFirst1] = useState(0)
  const [customRows1, setCustomRows1] = useState(10)
  const [customFirst2, setCustomFirst2] = useState(0)
  const [customRows2, setCustomRows2] = useState(10)
  const [customFirst3, setCustomFirst3] = useState(0)
  const [customRows3, setCustomRows3] = useState(10)
  const [contentFirst, setContentFirst] = useState(0)
  const [contentRow, setContentRows] = useState(10)
  const [currentPage, setCurrentPage] = useState(1)
  const [pageInputTooltip, setPageInputTooltip] = useState<string>(
    t('common_tooltip_goto_page')
  )

  const onBasicPageChange = (event: PaginatorPageState) => {
    setBasicFirst(event.first)
    setBasicRows(event.rows)
  }

  const onCustomPageChange1 = (event: PaginatorPageState) => {
    setCustomFirst1(event.first)
    setCustomRows1(event.rows)
    setCurrentPage(event.page + 1)
  }

  const onCustomPageChange2 = (event: PaginatorPageState) => {
    setCustomFirst2(event.first)
    setCustomRows2(event.rows)
  }

  const onCustomPageChange3 = (event: PaginatorPageState) => {
    setCustomFirst3(event.first)
    setCustomRows3(event.rows)
  }

  const onContentPageChange = (event: PaginatorPageState) => {
    setContentFirst(event.first)
    setContentRows(event.rows)
  }

  const onPageInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    setCurrentPage(Number(event.target.value))
  }

  const onPageInputKeyDown = (
    event: KeyboardEvent<HTMLInputElement>,
    option: PaginatorCurrentPageReportOptions
  ) => {
    if (event.key === 'Enter' && currentPage) {
      if (currentPage < 0 || currentPage > option.totalPages) {
        setPageInputTooltip &&
          setPageInputTooltip(
            t('common_tooltip_value_limited', {
              totalPages: option.totalPages,
            })
          )
      } else {
        const first = currentPage ? option.rows * (currentPage - 1) : 0
        setCustomFirst1(first)
        setPageInputTooltip &&
          setPageInputTooltip(t('common_tooltip_goto_page'))
      }
    }
  }

  const leftContent = (
    <Button
      type='button'
      icon='pi pi-refresh'
      onClick={() => setContentFirst(0)}
    />
  )
  const rightContent = <Button type='button' icon='pi pi-search' />

  const templateAlpha = TemplatePaginator(
    'sample_alpha',
    currentPage,
    pageInputTooltip,
    onPageInputKeyDown,
    onPageInputChange
  )

  const templateBeta = TemplatePaginator('sample_beta')
  const templateGamma = TemplatePaginator('sample_gamma')

  return (
    <div className='paginator-demo'>
      <div className='card'>
        <h5>Basic</h5>
        <Paginator
          first={basicFirst}
          rows={basicRows}
          totalRecords={120}
          rowsPerPageOptions={[10, 20, 30]}
          onPageChange={onBasicPageChange}
        ></Paginator>
        <br />
        <h5>Custom Template</h5>
        <Paginator
          template={templateAlpha}
          first={customFirst1}
          rows={customRows1}
          totalRecords={120}
          onPageChange={onCustomPageChange1}
        ></Paginator>
        <Paginator
          template={templateBeta}
          first={customFirst2}
          rows={customRows2}
          totalRecords={120}
          onPageChange={onCustomPageChange2}
          className='justify-content-end my-3'
        ></Paginator>
        <Paginator
          template={templateGamma}
          first={customFirst3}
          rows={customRows3}
          totalRecords={120}
          onPageChange={onCustomPageChange3}
          className='justify-content-start my-3'
        ></Paginator>
        <h5>Left and Right Content</h5>
        <Paginator
          first={contentFirst}
          rows={contentRow}
          totalRecords={12}
          onPageChange={onContentPageChange}
          leftContent={leftContent}
          rightContent={rightContent}
          template='FirstPageLink PrevPageLink CurrentPageReport NextPageLink LastPageLink'
        ></Paginator>
      </div>
    </div>
  )
}

export default PaginateGroup
