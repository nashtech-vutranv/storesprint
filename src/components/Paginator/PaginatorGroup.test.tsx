import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import PaginatorGroup from './PaginatorGroup'

test('renders PaginatorGroup', async () => {
  const tree = render(
    <MemoryRouter>
      <PaginatorGroup />
    </MemoryRouter>
  )
  
  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
})
