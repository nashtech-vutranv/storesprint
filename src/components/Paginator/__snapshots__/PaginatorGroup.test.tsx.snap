// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PaginatorGroup 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="paginator-demo"
      >
        <div
          class="card"
        >
          <h5>
            Basic
          </h5>
          <div
            class="p-paginator p-component"
          >
            <button
              class="p-paginator-first p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-double-left"
              />
            </button>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-left"
              />
            </button>
            <span
              class="p-paginator-pages"
            >
              <button
                class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
                type="button"
              >
                1
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                2
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                3
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                4
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link p-paginator-page-end"
                type="button"
              >
                5
              </button>
            </span>
            <button
              class="p-paginator-next p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-right"
              />
            </button>
            <button
              class="p-paginator-last p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-double-right"
              />
            </button>
            <div
              class="p-dropdown p-component p-inputwrapper p-inputwrapper-filled"
            >
              <div
                class="p-hidden-accessible"
              >
                <input
                  aria-haspopup="listbox"
                  readonly=""
                  type="text"
                />
              </div>
              <div
                class="p-hidden-accessible p-dropdown-hidden-select"
              >
                <select
                  aria-hidden="true"
                  tabindex="-1"
                >
                  <option
                    value=""
                  />
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
              </div>
              <span
                class="p-dropdown-label p-inputtext"
              >
                10
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="p-dropdown-trigger"
                role="button"
              >
                <span
                  class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
                />
              </div>
            </div>
          </div>
          <br />
          <h5>
            Custom Template
          </h5>
          <div
            class="p-paginator p-component"
          >
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-3"
              >
                organization_paginator_previous
              </span>
            </button>
            <span
              class="p-paginator-pages"
            >
              <button
                class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
                type="button"
              >
                1
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                2
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                3
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                4
              </button>
              <span
                class="p-paginator-page p-paginator-element p-link p-paginator-page-end p-disabled"
                style="user-select: none;"
              >
                ...
              </span>
            </span>
            <button
              class="p-paginator-next p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-3"
              >
                organization_paginator_next
              </span>
            </button>
            <div
              class="p-dropdown p-component p-inputwrapper p-inputwrapper-filled"
            >
              <div
                class="p-hidden-accessible"
              >
                <input
                  aria-haspopup="listbox"
                  readonly=""
                  type="text"
                />
              </div>
              <div
                class="p-hidden-accessible p-dropdown-hidden-select"
              >
                <select
                  aria-hidden="true"
                  tabindex="-1"
                >
                  <option
                    value=""
                  />
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
              </div>
              <span
                class="p-dropdown-label p-inputtext"
              >
                10
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="p-dropdown-trigger"
                role="button"
              >
                <span
                  class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
                />
              </div>
            </div>
            <span
              class="mx-3"
              style="user-select: none;"
            >
              organization_paginator_page_goto
              <input
                class="p-inputtext p-component p-filled ml-1"
                size="2"
                value="1"
              />
            </span>
          </div>
          <div
            class="p-paginator p-component justify-content-end my-3"
          >
            <span
              class="mx-1"
              style="user-select: none;"
            >
              organization_paginator_page_items
              :
               
            </span>
            <div
              class="p-dropdown p-component p-inputwrapper p-inputwrapper-filled"
            >
              <div
                class="p-hidden-accessible"
              >
                <input
                  aria-haspopup="listbox"
                  readonly=""
                  type="text"
                />
              </div>
              <div
                class="p-hidden-accessible p-dropdown-hidden-select"
              >
                <select
                  aria-hidden="true"
                  tabindex="-1"
                >
                  <option
                    value=""
                  />
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
              </div>
              <span
                class="p-dropdown-label p-inputtext"
              >
                10
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="p-dropdown-trigger"
                role="button"
              >
                <span
                  class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
                />
              </div>
            </div>
            <span
              style="user-select: none; width: 120px; text-align: center;"
            >
              1
               - 
              10
               of 
              120
            </span>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-left"
              />
            </button>
            <button
              class="p-paginator-next p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-right"
              />
            </button>
          </div>
          <div
            class="p-paginator p-component justify-content-start my-3"
          >
            <div
              class="d-flex align-items-center"
            >
              <span
                class="mr-3"
                style="user-select: none;"
              >
                organization_paginator_page_items
                :
                 
              </span>
              <div
                class="p-slider p-component slider p-slider-horizontal"
                style="width: 10rem;"
              >
                <span
                  class="p-slider-range"
                  style="width: 0%;"
                />
                <span
                  aria-valuemax="120"
                  aria-valuemin="10"
                  class="p-slider-handle"
                  role="slider"
                  style="left: 0%;"
                  tabindex="0"
                />
              </div>
            </div>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-left"
              />
            </button>
            <span
              class="p-paginator-pages"
            >
              <button
                class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
                type="button"
              >
                1
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                2
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                3
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link"
                type="button"
              >
                4
              </button>
              <button
                class="p-paginator-page p-paginator-element p-link p-paginator-page-end"
                type="button"
              >
                5
              </button>
            </span>
            <button
              class="p-paginator-next p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-right"
              />
            </button>
            <span
              style="user-select: none; width: 120px; text-align: center;"
            >
              1
               - 
              10
               of 
              120
            </span>
          </div>
          <h5>
            Left and Right Content
          </h5>
          <div
            class="p-paginator p-component"
          >
            <div
              class="p-paginator-left-content"
            >
              <button
                class="p-button p-component p-button-icon-only"
                type="button"
              >
                <span
                  class="p-button-icon p-c pi pi-refresh"
                />
                <span
                  class="p-button-label p-c"
                >
                   
                </span>
              </button>
            </div>
            <button
              class="p-paginator-first p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-double-left"
              />
            </button>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-left"
              />
            </button>
            <span
              class="p-paginator-current"
            >
              (1 of 2)
            </span>
            <button
              class="p-paginator-next p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-right"
              />
            </button>
            <button
              class="p-paginator-last p-paginator-element p-link"
              type="button"
            >
              <span
                class="p-paginator-icon pi pi-angle-double-right"
              />
            </button>
            <div
              class="p-paginator-right-content"
            >
              <button
                class="p-button p-component p-button-icon-only"
                type="button"
              >
                <span
                  class="p-button-icon p-c pi pi-search"
                />
                <span
                  class="p-button-label p-c"
                >
                   
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="paginator-demo"
    >
      <div
        class="card"
      >
        <h5>
          Basic
        </h5>
        <div
          class="p-paginator p-component"
        >
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-double-left"
            />
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-left"
            />
          </button>
          <span
            class="p-paginator-pages"
          >
            <button
              class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
              type="button"
            >
              1
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              2
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              3
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              4
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link p-paginator-page-end"
              type="button"
            >
              5
            </button>
          </span>
          <button
            class="p-paginator-next p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-right"
            />
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-double-right"
            />
          </button>
          <div
            class="p-dropdown p-component p-inputwrapper p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
        </div>
        <br />
        <h5>
          Custom Template
        </h5>
        <div
          class="p-paginator p-component"
        >
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_previous
            </span>
          </button>
          <span
            class="p-paginator-pages"
          >
            <button
              class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
              type="button"
            >
              1
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              2
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              3
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              4
            </button>
            <span
              class="p-paginator-page p-paginator-element p-link p-paginator-page-end p-disabled"
              style="user-select: none;"
            >
              ...
            </span>
          </span>
          <button
            class="p-paginator-next p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_next
            </span>
          </button>
          <div
            class="p-dropdown p-component p-inputwrapper p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
          <span
            class="mx-3"
            style="user-select: none;"
          >
            organization_paginator_page_goto
            <input
              class="p-inputtext p-component p-filled ml-1"
              size="2"
              value="1"
            />
          </span>
        </div>
        <div
          class="p-paginator p-component justify-content-end my-3"
        >
          <span
            class="mx-1"
            style="user-select: none;"
          >
            organization_paginator_page_items
            :
             
          </span>
          <div
            class="p-dropdown p-component p-inputwrapper p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
          <span
            style="user-select: none; width: 120px; text-align: center;"
          >
            1
             - 
            10
             of 
            120
          </span>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-left"
            />
          </button>
          <button
            class="p-paginator-next p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-right"
            />
          </button>
        </div>
        <div
          class="p-paginator p-component justify-content-start my-3"
        >
          <div
            class="d-flex align-items-center"
          >
            <span
              class="mr-3"
              style="user-select: none;"
            >
              organization_paginator_page_items
              :
               
            </span>
            <div
              class="p-slider p-component slider p-slider-horizontal"
              style="width: 10rem;"
            >
              <span
                class="p-slider-range"
                style="width: 0%;"
              />
              <span
                aria-valuemax="120"
                aria-valuemin="10"
                class="p-slider-handle"
                role="slider"
                style="left: 0%;"
                tabindex="0"
              />
            </div>
          </div>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-left"
            />
          </button>
          <span
            class="p-paginator-pages"
          >
            <button
              class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
              type="button"
            >
              1
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              2
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              3
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              4
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link p-paginator-page-end"
              type="button"
            >
              5
            </button>
          </span>
          <button
            class="p-paginator-next p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-right"
            />
          </button>
          <span
            style="user-select: none; width: 120px; text-align: center;"
          >
            1
             - 
            10
             of 
            120
          </span>
        </div>
        <h5>
          Left and Right Content
        </h5>
        <div
          class="p-paginator p-component"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              class="p-button p-component p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                 
              </span>
            </button>
          </div>
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-double-left"
            />
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-left"
            />
          </button>
          <span
            class="p-paginator-current"
          >
            (1 of 2)
          </span>
          <button
            class="p-paginator-next p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-right"
            />
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-angle-double-right"
            />
          </button>
          <div
            class="p-paginator-right-content"
          >
            <button
              class="p-button p-component p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-search"
              />
              <span
                class="p-button-label p-c"
              >
                 
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
